src/sympc/protocol/spdz/spdz.py

```diff
@@ -33,8 +40,12 @@ def mul_master(x, y, op_str):
     eps_plaintext = eps.reconstruct(decode=False)
     delta_plaintext = delta.reconstruct(decode=False)
 
-    with ThreadPoolExecutor(max_workers=nr_parties) as executor:
-        args = list(zip(session.session_ptr, a_sh.shares, b_sh.shares, c_sh.shares))
+    with ThreadPoolExecutor(
+        max_workers=nr_parties, thread_name_prefix="spdz_mul_master"
+    ) as executor:
```

Comment:
Idea for later: if we package this in a decorator we might end up with a super sexy syntax

Reply:
Really like this idea - will do it here.